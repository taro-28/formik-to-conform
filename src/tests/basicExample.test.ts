import { convert } from "..";
import { test, expect } from "vitest";
import { readFileSync } from "node:fs";
import path from "node:path";

test("basic example", () => {
  const file = readFileSync(
    path.join(__dirname, "formik.basicExample.jsx"),
    "utf-8",
  );
  expect(convert(file)).toMatchSnapshot();
});
