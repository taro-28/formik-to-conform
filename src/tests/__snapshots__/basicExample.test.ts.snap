// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic example 1`] = `
"import React from 'react';
import ReactDOM from 'react-dom';
import { useForm, conform, type SubmitHandler } from '@conform-to/react';

export const Basic = () => {
  const [form, { firstName, lastName, email }] = useForm({
    defaultValue: {
      firstName: "",
      lastName: "",
      email: "",
    },
    onSubmit: async ({ formData }) => {
      const data = Object.fromEntries(formData) as Values;
      await new Promise((r) => setTimeout(r, 500));
      alert(JSON.stringify(data, null, 2));
    },
  });

  return (
    <div>
      <h1>Sign Up</h1>
      <form {...form.props}>
        <label htmlFor="firstName">First Name</label>
        <input
          {...conform.input(firstName)}
          id="firstName"
          placeholder="Jane"
        />

        <label htmlFor="lastName">Last Name</label>
        <input
          {...conform.input(lastName)}
          id="lastName"
          placeholder="Doe"
        />

        <label htmlFor="email">Email</label>
        <input
          {...conform.input(email)}
          id="email"
          type="email"
          placeholder="jane@acme.com"
        />

        <button type="submit">Submit</button>
      </form>
    </div>
  );
}

ReactDOM.render(<Basic />, document.getElementById('root'));
"
`;
